src/programs/federal/snap.js

```diff
@@ -288,39 +267,32 @@ hlp.monthlyNetIncome = function(client) {
     return Math.max( 0, afterDeductions );
 };
 
-hlp.maxTotalNetMonthlyIncome = function (client) {
-    var maxTotalNetIncome = null;
-    //TODO: Logic different in website calculate; when (hlp.monthlyNetIncome < 0 ) = 0 while excel return a number
-    if ( hlp.isNetIncomeTest(client)===false ) {
-      maxTotalNetIncome = "no limit";
-      return maxTotalNetIncome;
-    } else {
-      return getYearlyLimitBySize(SNAPData.NET_INCOME_LIMITS, hlp.householdSize( client ));
-    }
+hlp.getMaxNetIncome = function (client) {
+  //TODO: Logic different in website calculate; when (hlp.monthlyNetIncome < 0 ) = 0 while excel return a number
+  var totalMonthlyGross       = hlp.getTotalMonthlyGross(client),
+      povertyGrossIncomeLevel = hlp.getPovertyGrossIncomeLevel(client),
+      disabledOrElderlyMember = hlp.hasDisabledOrElderlyMember(client);
+  
+  if ( (totalMonthlyGross <= povertyGrossIncomeLevel) || !disabledOrElderlyMember ) {
+    return "no limit";
+  } else {
+    return getYearlyLimitBySize(SNAPData.NET_INCOME_LIMITS, hlp.householdSize( client ));
+  }
 };
 
 // NET INCOME TEST RESULT
 hlp.getNetIncomeTestResult = function(client ) {
-  if ( hlp.maxTotalNetMonthlyIncome(client) === "no limit" ) {
-      return true;
-    } else if ( hlp.monthlyNetIncome(client) < hlp.maxTotalNetMonthlyIncome(client) ) {
-      return true;
-    }
-    else {
-      return false;
-    }
-};
 
-// FINAL DETERMINATION
-hlp.getThirtyPercentNetIncome = function(client) {
-  if ( hlp.monthlyNetIncome(client) * SNAPData.PERCENT_OF_NET > 0 ) {
-    return hlp.monthlyNetIncome(client) * SNAPData.PERCENT_OF_NET;
+  var maxNetIncome = hlp.getMaxNetIncome(client)
+
+  if ( maxNetIncome === "no limit" ) {
+    return true;
+  } else if ( hlp.monthlyNetIncome(client) < maxNetIncome ) {
+    return true;
+  } else {
+    return false;
```

Comment:
Not a fan of `if(true) return true;`. I always assume the extra code is there for some reason I'm not seeing, and end up rereading it a few times.

But that could just be me.

Reply:
No, I hear you there. I'm not fixing everything at once otherwise it's going to be hard to examine changes. I think I've already fixed that in one of my more recent edits.